cardano-db-sync/src/Cardano/DbSync/Config.hs

```diff
@@ -88,3 +94,61 @@ coalesceConfig pcfg ncfg adjustGenesisPath = do
 
 mkAdjustPath :: SyncPreConfig -> (FilePath -> FilePath)
 mkAdjustPath cfg fp = takeDirectory (pcNodeConfigFilePath cfg) </> fp
+
+-- check both whitelist but also checking plutus Maybes first
+-- TODO: cmdv: unsure if this is correct because if plutusMaybeCheck fails then no multiasset whitelist is not checked
+plutusMultiAssetWhitelistCheck :: SyncEnv -> [Generic.TxOut] -> Bool
+plutusMultiAssetWhitelistCheck syncEnv txOuts =
+  plutusMaybeCheck txOuts && (plutusWhitelistCheck syncEnv txOuts || multiAssetWhitelistCheck syncEnv txOuts)
+
+plutusMaybeCheck :: [Generic.TxOut] -> Bool
+plutusMaybeCheck =
+  any (\txOut -> isJust (Generic.txOutScript txOut) || isJust (Generic.maybePaymentCred $ Generic.txOutAddress txOut))
+
+plutusWhitelistCheck :: SyncEnv -> [Generic.TxOut] -> Bool
+plutusWhitelistCheck syncEnv txOuts = do
+  -- first check the config option
+  case ioPlutusExtra iopts of
+    PlutusEnable -> True
+    PlutusDisable -> False
```

Comment:
As we discussed Disabled should return `True` at this point, conceptually meaning that the tx is in fact whitelisted. Later, while inserting the plutus data, it will be checked again.